File: Mvp_java_b/app/src/main/java/com/example/tonjies/mvp_java_b/module/info/InfoActivity.java
package com.example.tonjies.mvp_java_b.module.info;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import com.example.tonjies.mvp_java_b.R;
import com.example.tonjies.mvp_java_b.module.info.prsenter.InfoPresenter;
import com.example.tonjies.mvp_java_b.module.info.prsenter.contract.InfoContract;

import org.jetbrains.annotations.Contract;
/**
 * 获取学生信息Activity
 */
public class InfoActivity extends AppCompatActivity implements InfoContract.infoView{
    private Button button;

    //声明P层变量
    private InfoPresenter infoPresenter;

    private TextView textView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_info);
        infoPresenter=new InfoPresenter(this);
        button=findViewById(R.id.btn_01);//点击请求
        textView=findViewById(R.id.txt_01);//显示文本
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                infoPresenter.getData();//进行网络请求
            }
        });
    }

    /**
     * 在P层中回调的方法
     * @param msg
     */
    @Override
    public void show(String msg) {
        textView.setText("获取到的学生数据:"+msg);
    }

    /**
     * 销毁View对象
     */
    @Override
    protected void onDestroy() {
        super.onDestroy();
        infoPresenter.detachView();
    }
}

File: Mvp_kotlins_a/app/src/main/java/com/example/mvp_kotlins_a/module/info/InfoActivity.kt
package com.example.mvp_kotlins_a.module.info

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import com.example.mvp_kotlins_a.R
import com.example.mvp_kotlins_a.module.info.presenter.InfoPresenter
import com.example.mvp_kotlins_a.module.info.view.InfoView
import kotlinx.android.synthetic.main.activity_info.*

/**
 * Created by 舍长
 * describe:在P层获取完网络数据后显示在界面上
 */
class InfoActivity : AppCompatActivity(), InfoView {

    //声明p层接口
    private var infoPresenter = InfoPresenter()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_info)
        //完成对view接口的绑定，此时p层还没有该方法
        infoPresenter.attachView(this)
        btnSend.setOnClickListener {
            infoPresenter.getData()
        }
    }

    /**
     * O层返回数据的回调方法
     */
    override fun show(msg: String) {
        //将请求到的数据设置到TextView上
        txtStudentInfo.text = msg
    }

    override fun onDestroy() {
        super.onDestroy()
        //销毁P层，防止内存泄漏，同样，此时还有没该方法
        infoPresenter.detechView()
    }
}
